File: backend/Models/Recipe.cs
using backend.Models.Ingredients;

namespace backend.Models
{
    public class Recipe
    {
        public int Id { get; set; }
        public string? Name { get; set; }
        public string? Description { get; set; }
        public string? Category { get; set; }
        public string? Area { get; set; }
        public string? YoutubeVideo { get; set; }
        public ICollection<RecipeIngredient>? RecipeIngredients { get; set; }
    }
}

File: icare.client/src/models/recipe.ts
export { Recipe, RecipeIngredient }

interface Recipe {
  id: number;
  picture: string;
  name: string;
  category: string,
  area: string,
  instructions: string;
  ingredients: RecipeIngredient[];
  urlVideo: string;
  isFavorite: boolean;
  calories: number;
  proteins: number;
  carbohydrates: number;
  lipids: number;
  fibers: number;
}

interface RecipeIngredient {
  name: string;
  measure: string;
  grams: number;
}
